source/extensions/filters/http/adaptive_concurrency/controller/gradient_controller.h

```diff
@@ -283,6 +289,9 @@ class GradientController : public ConcurrencyController {
 
   Event::TimerPtr min_rtt_calc_timer_;
   Event::TimerPtr sample_reset_timer_;
+
+  // Used for testing only.
+  mutable Thread::ThreadSynchronizer synchronizer_;
```

Comment:
nit: Does this member need to be mutable?  Seems like it only used in non-const methods.

Reply:
thanks, updated.